src/lib.rs

```diff
@@ -160,8 +161,86 @@ use bitcoin_hashes::Hash;
 
 /// An ECDSA signature
 #[derive(Copy, Clone, PartialEq, Eq)]
-pub struct Signature(ffi::Signature);
+#[cfg_attr(feature="json", derive(schemars::JsonSchema))]
+pub struct Signature(
+    #[cfg_attr(feature="json", schemars(schema_with="schemas::signature"))]
+    ffi::Signature);
+
+/// Custom JSON Schemas for interior types
+#[cfg(feature="json")]
+pub mod schemas {
+    use schemars::schema::{Schema, SchemaObject};
+    use schemars::{gen::SchemaGenerator, JsonSchema};
+    const HEX_REGEX : &'static str = "[0-9a-fA-F]+";
```

Comment:
```suggestion
    const HEX_REGEX : &'static str = "([0-9a-fA-F][0-9a-fA-F])*";
```

Reply:
Good catch, i use this form some other places I should be consistent to make sure it matches full bytes only.